### DISCUSSION TITLE
Spotlight / Command Palette alike tool

### DISCUSSION THREAD

**Discussion by joselcvarela** [directus/directus#8408] [Feature Requests]
I think a tool like Spotlight from OSX or Command Palette from VSCode would increase the productivity a lot.

You just press Ctrl/Cmd + Space or click an icon, and you could:
- Navigate to anywhere in the app, from items in collections to items in data model;
- Create Field, Export CSV, Upload file, and other relevant commands;
- Compute results based on formula (like Excel)
- Bookmark entries

**Comment by azrikahar** [comment]
Interesting! Never thought of it in the context of Directus. This also reminded me of the recent release by Logseq which adds this Command Palette: https://github.com/logseq/logseq/releases/tag/0.3.7 (first gif in the link)

**Reply by benhaynes** [reply]
Great one, @joselcvarela!! I love this idea. Let's use this ticket to brainstorm all the different functions of this Command Palette, and ensure each command can be supported (maybe some require you to be on a specific page?). Some thoughts below:

* I like the `META-SPACE` shortcut (or similar) to open the command menu, and/or `/` to open search (this could be reversed)
* The default search would recursively run a full-text `search` across all collections (perhaps with an option to limit to the current collection). This might be a _heavy_ set of requests, so we'll have to think about how we can optimize. One idea I had (that would help elsewhere too) would be to create a `directus_search_index` collection that holds a global lookup cache. The main benefit here is that we can concatenate relational values into the searchable string for better results.
* For the command menu, you could get the command menu with autocomplete. A helper list of common options should be displayed like in the link from @azrikahar. Some options:
    * Enter full-screen mode (all pages)
    * Go to User Profile
    * Export current/selected items
    * Toggle query log (could show all API queries running in the app background)
    * Toggle Light/Dark mode
    * Show help menu and all keyboard shortcuts

It'd be cool if we could figure out a good command _chaining_ syntax, so you could enter the command palette, press a shortcut key, and then type to search/autocomplete within that. For example:

1. `META-SPACE` — open command palette
2. `b` — "Bookmarks" shortcut
3. `vip` — Autocomplete "VIP Customers" bookmark
4. `ENTER` — go to that bookmark

So many ideas!

**Reply by joselcvarela** [reply]
We could also differentiate commands from search.

For example, on VS Code, Search and Commands are distinguished by a `>` on the start.

![8QVxbUTL7D](https://user-images.githubusercontent.com/14039341/135279545-f152de48-da56-44b2-ab99-7f5311645262.gif)

**Reply by rijkvanzanten** [reply]
(Just to add my 2¢ off the top: it shouldn't be cmd+space, as that'd override the system spotlight search which would be super annoying)

**Reply by benhaynes** [reply]
When I say `META`, I am usually talking about the _secondary_ key (ALT/OPTION on Mac), since COMMAND is always a tricky one to override.

**Reply by rijkvanzanten** [reply]
Gotcha. FWIW "meta" in the context of the app codebase is used as the identifier for _either_ Ctrl or Command based on your OS 👍🏻

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by joselcvarela** [reply]
But boss said he loved that 😭
Just kidding, maybe one day 😅
